app/gui2/src/components/visualizations/TableVisualization.vue

```diff
@@ -265,12 +270,40 @@ function isMatrix(data: object): data is LegacyMatrix {
   return json.every((d) => d.length === firstLen)
 }
 
-function toField(name: string): ColDef {
-  return { field: name }
+function toField(name: string, valType?: string): ColDef {
+  let icon
+  switch (valType) {
+    case 'Char':
+      icon = 'text'
+      break
+    case 'Boolean':
+      icon = 'check'
+      break
+    case 'Integer':
+    case 'Float':
+      icon = 'math'
+      break
```

Comment:
We are also missing cases for types:
- `Decimal`
- `Date_Time`
- `Time_Of_Day` (maybe some clock could work?)
- `Byte`

Are they supposed to be added later?

Also I'd like to make sure (at least by manual testing) what happens if a column with such not-yet handled type is visualized? Just to ensure the visualization does not break in that case.

Reply:
I currently don't have 2 icons in my icon set. Are there tooltips that show the full type name?

Reply:
Just to visually show you what would happen (before having added any kind of Time type if a time column came in and there was no icon, it could just show the title:
<img width="497" alt="image" src="https://github.com/enso-org/enso/assets/170310417/8062aa92-cc26-474e-926e-f4136fedf9f5">

Reply:
@AdRiley I have added the ones that Radek has suggested but wanted to check with you if there is a clock/ time icon, I can't see one in the list- if not what would be the most appropriate for a time type?

Reply:
There is time and time2 
![image](https://github.com/enso-org/enso/assets/1720119/6a887c34-bdf6-40ed-a598-1f2334e1d430)

Reply:
> I currently don't have 2 icons in my icon set. Are there tooltips that show the full type name?

This would be great. I think ideally the tooltip should show the `to_display_text` representation of the type, e.g. it would show more details like how many bits or if the text is variable length or fixed length: `Char (variable length, max_size=unlimited)` or `Integer (16 bits)`.